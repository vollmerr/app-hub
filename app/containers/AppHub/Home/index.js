import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import styled from 'styled-components';

import toJS from '../../../hocs/toJS';
import theme from '../../../utils/theme';
import { escapeRegExp } from '../../../utils/string';

import * as selectors from '../selectors';

import Search from './Search';
import AppTiles from './AppTiles';
import MessageSection from './MessageSection';


export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: calc(100vh - ${theme.hub.headerHeight}px);
`;


export const Container = styled.div`
  align-items: center;
  justify-content: center;
  display: flex;
  flex-direction: column;
  padding: 15px;
  width: ${theme.breakpoints.xs - 10}px;
  @media (min-width: ${theme.breakpoints.sm}px) {
    width: ${theme.breakpoints.sm - 10}px;
  }
  @media (min-width: ${theme.breakpoints.md}px) {
    width: ${theme.breakpoints.md - 10}px;
  }
  @media (min-width: ${theme.breakpoints.lg}px) {
    width: ${theme.breakpoints.lg - 10}px;
  }
  @media (min-width: ${theme.breakpoints.xl}px) {
    width: ${theme.breakpoints.xl - 10}px;
  }
`;


export const UpperSection = styled.div`
  align-items: center;
  justify-content: center;
  display: flex;
  flex-direction: column;
  flex: 1;
  background: ${theme.themeDarker};
  min-height: 300px;
`;


export const LowerSection = UpperSection.extend`
  background: ${theme.neutralLight};
  padding: ${(props) => props.isMobile ? '0' : '15px'};
`;


export class Home extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      appRoutes: this.props.user.routes.slice(1), // remove 'AppHub' Tile
    };
  }

  /**
   * Filters list of apps being displayed
   */
  handleChangeSearch = (value) => {
    const { user } = this.props;
    const re = new RegExp(escapeRegExp(value), 'i');
    // remove 'AppHub' route then filter based off search input
    const appRoutes = user.routes.slice(1).filter((route) => {
      const { title, desc, keywords } = route.meta;
      const toFilter = `${title} ${desc} ${keywords} ${route.name}`;
      return toFilter.match(re);
    });

    this.setState({ appRoutes });
  }

  render() {
    const { view } = this.props;
    const { appRoutes } = this.state;
    const { isMobile } = view;

    return (
      <Wrapper>
        <UpperSection>
          <Container>
            <Search isMobile={isMobile} onChange={this.handleChangeSearch} />
            <AppTiles routes={appRoutes} />
          </Container>
        </UpperSection>
        <LowerSection isMobile={isMobile}>
          <MessageSection isMobile={isMobile} />
        </LowerSection>
      </Wrapper>
    );
  }
}


const { object } = PropTypes;

Home.propTypes = {
  view: object.isRequired,
  user: object.isRequired,
};


const mapStateToProps = createStructuredSelector({
  view: selectors.getView,
  user: selectors.getUser,
});

const withConnect = connect(mapStateToProps);


export default compose(
  withConnect,
  toJS,
)(Home);
