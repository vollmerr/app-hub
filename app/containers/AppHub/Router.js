import React from 'react';
import { Switch, Route } from 'react-router-dom';

import AppHubHome from 'containers/AppHub-Home/Loadable';
import Spa from 'containers/Spa/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';

export const routes = [
  {
    key: 'home',
    name: 'App Hub',
    path: '/',
    exact: true,
    component: AppHubHome,
    icon: 'Glimmer',
  },
  {
    key: 'spa',
    name: 'SPA',
    path: '/spa',
    exact: false,
    component: Spa,
    icon: 'BulletedList2Mirrored',
  },
  {
    key: 'cold',
    name: 'Cold',
    href: 'https://cold.govops.ca.gov/',
    icon: 'Rename',
  },
];

export const notFound = NotFoundPage;

const Router = () => (
  <Switch>
    {
      routes.map((route) => (
        <Route exact={route.exact} key={route.path} path={route.path} component={route.component} />
      ))
    }
    <Route component={notFound} />
  </Switch>
);

export default Router;
