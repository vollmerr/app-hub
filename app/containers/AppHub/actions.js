/*
 *
 * AppHub actions
 *
 */

import {
  CHANGE_MOBILE,
} from './constants';

export function changeMobile(isMobile) {
  return {
    isMobile,
    type: CHANGE_MOBILE,
  };
}
