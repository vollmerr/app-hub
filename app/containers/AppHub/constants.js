// VIEW
export const CHANGE_MOBILE = 'AppHub/CHANGE_MOBILE';
export const CHANGE_PANEL_OPEN = 'AppHub/CHANGE_PANEL_OPEN';
export const CHANGE_PANEL_SELECTED = 'AppHub/CHANGE_PANEL_SELECTED';
// panel names
export const HELP_PANEL = 'AppHub/HELP_PANEL';
export const APPS_PANEL = 'AppHub/APPS_PANEL';
export const APP_NAV_PANEL = 'AppHub/APP_NAV_PANEL';
export const ALERTS_PANEL = 'AppHub/ALERTS_PANEL';
export const DEV_PANEL = 'AppHub/DEV_PANEL';
// APP
export const CHANGE_APP = 'AppHub/CHANGE_APP';
export const CHANGE_APP_STATUS = 'AppHub/CHANGE_APP_STATUS';
// user
export const AUTH_USER = 'AppHub/AUTH_USER';
export const AUTH_USER_DONE = 'AppHub/AUTH_USER_DONE';
// REQUEST STATUS TYPES
export const REQUEST = '_REQUEST';
export const SUCCESS = '_SUCCESS';
export const FAILURE = '_FAILURE';

export const COL_TYPES = {
  DATE: 'DATE',
};

export const ROLES = {
  USER: 'AppHub User',
};

// D3
export const INCREMENT_RENDER_COUNT = 'AppHub/INCREMENT_RENDER_COUNT';
