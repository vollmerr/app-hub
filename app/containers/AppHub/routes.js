import AppHubHome from 'containers/AppHub-Home';
import Spa from 'containers/Spa/Loadable';

import Loading from 'components/Loading';

import { meta } from './meta';

const routes = [
  {
    key: 'home',
    name: 'App Hub',
    path: '/',
    exact: true,
    component: AppHubHome,
    icon: 'Glimmer',
    meta: meta.apphub,
  },
  {
    key: 'spa',
    name: 'SPA',
    path: '/spa',
    exact: false,
    component: Spa,
    icon: 'BulletedList2Mirrored',
    meta: meta.spa,
  },
  {
    key: 'loading',
    name: 'Test App Loading',
    path: '/loading',
    component: Loading,
    icon: 'ProgressRingDots',
    meta: meta.loading,
  },
  {
    key: 'cold',
    name: 'COLD',
    href: 'https://cold.govops.ca.gov/',
    icon: 'Rename',
    meta: meta.cold,
  },
  {
    key: 'bars',
    name: 'BARS',
    href: 'http://bars.technology.ca.gov/',
    icon: 'IDBadge',
    meta: meta.bars,
  },
  {
    key: 'ed',
    name: 'ED',
    href: 'http://employeedirectory/',
    icon: 'ContactInfo',
    meta: meta.ed,
  },
];

export const apps = routes.slice(1);

export default routes;
