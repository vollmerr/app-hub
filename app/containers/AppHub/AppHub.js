/**
 *
 * AppHub
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
// import AppHubHeader from 'containers/AppHub-Header';
import AppHubHeader from 'components/AppHub-Header';
import AppHubPanel from 'components/AppHub-Panel';
import theme from 'utils/theme';

import injectSaga from 'utils/injectSaga';
// import injectReducer from 'utils/injectReducer';
import {
  makeSelectIsMobile,
  makeSelectPanelIsOpen,
  makeSelectPanelSelected,
  makeSelectAppRoutes,
  makeSelectAppMeta,
  makeSelectUserSam,
} from './selectors';
// import reducer from './reducer';
import saga from './saga';
import { changeMobile, changePanelOpen, changePanelSelected, authUserRequest } from './actions';

export class AppHub extends React.Component { // eslint-disable-line react/prefer-stateless-function
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    this.props.dispatch(authUserRequest());
    this.handleResize();
    window.addEventListener('resize', this.handleResize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
  }

  handleResize = () => {
    const { dispatch, isMobile } = this.props;
    const mobile = window.innerWidth <= theme.breakpoints.lg;

    if (mobile !== isMobile) {
      dispatch(changeMobile(mobile));
    }
  }

  handlePanelClick = (panel) => {
    const { dispatch, panelSelected, panelIsOpen } = this.props;
    const selected = panel || panelSelected; // errors if null, so set a default
    // click on same panel or overlay
    if (!panel || (panelSelected === panel && panelIsOpen)) {
      // close it
      dispatch(changePanelOpen(false));
    } else {
      // open the panel (don't dispatch unessecary actions if already open)
      if (!panelIsOpen) {
        dispatch(changePanelOpen(true));
      }
      dispatch(changePanelSelected(selected));
    }
  }

  render() {
    const { isMobile, panelSelected, panelIsOpen, appMeta, appRoutes, userName } = this.props;

    const headerProps = {
      isMobile,
      userName,
      panel: panelSelected,
      isOpen: panelIsOpen,
      appName: appMeta.name,
      appPath: appRoutes.find((route) => route.key === 'home').path,
      onClick: this.handlePanelClick,
    };

    const panelProps = {
      routes: appRoutes,
      panel: panelSelected,
      isOpen: panelIsOpen,
      onClick: this.handlePanelClick,
    };

    return (
      <div>
        <AppHubHeader {...headerProps} />
        <AppHubPanel {...panelProps} />
      </div>
    );
  }
}

AppHub.propTypes = {
  dispatch: PropTypes.func.isRequired,
  isMobile: PropTypes.bool.isRequired,
  panelIsOpen: PropTypes.bool.isRequired,
  panelSelected: PropTypes.string.isRequired,
  // appName: PropTypes.string,
  // appPath: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  isMobile: makeSelectIsMobile(),
  panelIsOpen: makeSelectPanelIsOpen(),
  panelSelected: makeSelectPanelSelected(),
  appRoutes: makeSelectAppRoutes(),
  appMeta: makeSelectAppMeta(),
  userName: makeSelectUserSam(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);

// const withReducer = injectReducer({ key: 'appHub', reducer });
const withSaga = injectSaga({ key: 'appHub', saga });

export default compose(
  // withReducer,
  withSaga,
  withConnect,
)(AppHub);
