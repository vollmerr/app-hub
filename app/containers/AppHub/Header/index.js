import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import toJS from '../../../hocs/toJS';
import theme from '../../../utils/theme';

import meta from '../meta';
import * as selectors from '../selectors';
import * as actions from '../actions';
import * as C from '../constants';

import NavLink from './NavLink';
import Logo from './Logo';
import UserInfo from './UserInfo';


const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  height: ${theme.hub.headerHeight}px;
  box-sizing: border-box;
  background: ${theme.neutralDark};
`;


const Section = styled.div`
  display: flex;
`;


const Line = styled.div`
  width: 1px;
  margin: auto;
  background: ${theme.neutralSecondary};
  height: ${(props) => props.partial ?
    `calc(${theme.hub.headerHeight}px - 15px)` :
    `calc(${theme.hub.headerHeight}px - 2px)`};
`;


class Header extends React.PureComponent {
  /**
   * Handles clicking on a panel from the header, determines if should open or close
   */
  handlePanelClick = (name) => {
    const { onChangePanelOpen, onChangePanelSelected, panel } = this.props;
    const selected = name || panel.selected; // errors if null, so set a default
    // click on same panel or overlay
    if (!name || (panel.selected === name && panel.isOpen)) {
      // close it
      onChangePanelOpen(false);
    } else {
      // open the panel (don't dispatch unessecary actions if already open)
      if (!panel.isOpen) {
        onChangePanelOpen(true);
      }
      onChangePanelSelected(selected);
    }
  }

  /**
   * Renders a nav link
   * @param {object} props    - props to pass through to nav link
   *
   * @return {JSX}            - NavLink to render
   */
  renderLink = (props) => {
    const { panel } = this.props;
    const linkProps = {
      onClick: this.handlePanelClick,
      ...props,
    };

    if (panel) {
      linkProps.checked = props.panel === panel.selected && panel.isOpen;
    }

    return (
      <NavLink {...linkProps} />
    );
  }

  render() {
    const {
      app,
      user,
      view,
    } = this.props;

    const { name: userName } = user;
    const { name: appName } = app;
    const { isMobile } = view;

    const appMenu = {
      isVisible: isMobile && appName,
      props: {
        iconProps: {
          iconName: 'appMenu',
        },
        title: `${appName} Navigation`,
        panel: C.APP_NAV_PANEL,
      },
    };

    const appHubTitle = {
      isVisible: !isMobile || !appName,
      props: {
        to: '/',
        text: meta.title,
        title: meta.title,
        iconProps: {
          iconName: 'Apphub',
          style: { fontSize: '24px', marginRight: '10px' },
        },
        padding: '15px',
      },
    };

    const appTitle = {
      isVisible: appName,
      props: {
        to: 'app.path', // TODO!
        text: appName,
        title: appName,
        padding: '15px',
      },
    };

    const userInfo = {
      isVisible: !isMobile && userName,
      props: {
        name: userName,
        initials: userName ? userName.split(/[,. ]/, 2).map((x) => x[0].toUpperCase()).join('') : '',
      },
    };

    const helpPanelProps = {
      iconProps: {
        iconName: 'helpMenu',
      },
      title: 'Help Panel',
      panel: C.HELP_PANEL,
    };

    const appHubPanelProps = {
      dark: true,
      iconProps: {
        iconName: 'appHubMenu',
      },
      title: 'App Hub Navigation',
      panel: C.APPS_PANEL,
    };

    return (
      <Wrapper>
        <Section>
          {appMenu.isVisible && this.renderLink(appMenu.props)}

          {!isMobile && <Logo />}
          {!isMobile && <Line partial />}

          {appHubTitle.isVisible && this.renderLink(appHubTitle.props)}
          {!name && <Line partial={!isMobile} />}

          {appTitle.isVisible && this.renderLink(appTitle.props)}
        </Section>

        <Section>
          {userInfo.isVisible && <UserInfo {...userInfo.props} />}
          <Line />

          {/* TODO: DEV/MOCK PANELS (+ Line) */}

          {/* TODO: ALERTS PANEL (+ Line) */}

          {this.renderLink(helpPanelProps)}
          <Line />

          {this.renderLink(appHubPanelProps)}
        </Section>
      </Wrapper>
    );
  }
}


const { func, object } = PropTypes;

Header.propTypes = {
  app: object.isRequired,
  user: object.isRequired,
  view: object.isRequired,
  panel: object.isRequired,
  onChangePanelOpen: func.isRequired,
  onChangePanelSelected: func.isRequired,
};


const mapStateToProps = createStructuredSelector({
  app: selectors.getApp,
  user: selectors.getUser,
  view: selectors.getView,
  panel: selectors.getViewPanel,
});

const mapDispatchToProps = (dispatch) => ({
  onChangePanelOpen: (open) => dispatch(actions.changePanelOpen(open)),
  onChangePanelSelected: (panel) => dispatch(actions.changePanelSelected(panel)),
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);


export default compose(
  withConnect,
  toJS,
)(Header);
