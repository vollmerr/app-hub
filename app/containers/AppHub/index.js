import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { Fabric } from 'office-ui-fabric-react/lib/Fabric';

import Header from './Header';
import * as selectors from './selectors';


export const Wrapper = styled(Fabric) `
  height: 100%;
  overflow: hidden;
`;


export class AppHub extends React.PureComponent {
  render() {
    return (
      <Wrapper>
        <Header />
        {
          isAuthenticated ?
            <Router routes={routes} /> :
            <Loading pastDelay {...this.props.app} />
        }
      </Wrapper>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  isAuthenticated: selectors.getIsAuthenticated,
});


export default compose(
  withSaga,
  withConnect,
)(AppHub);

// tests....
