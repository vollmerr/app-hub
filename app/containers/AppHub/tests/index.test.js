import React from 'react';
import { shallow } from 'enzyme';
import { Fabric } from 'office-ui-fabric-react/lib/Fabric';

import LoadingMessage from 'components/Loading/LoadingMessage';
import { testStyledComponent } from 'utils/testUtils';
import Router from 'components/Router';

import AppHubRoot, { Wrapper, Content } from '../index';
import AppHub from '../AppHub';
import routes from '../routes';

testStyledComponent(Wrapper, Fabric);
testStyledComponent(Content);


describe('<AppHubRoot />', () => {
  let wrapper;
  beforeEach(() => {
    localStorage.setItem = jest.fn();
    localStorage.getItem = jest.fn();
    wrapper = shallow(<AppHubRoot />);
  });

  it('should render correctly if not authenticated', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly if authenticated', () => {
    wrapper.setState({ isAuthenticated: true });
    expect(wrapper).toMatchSnapshot();
  });


  it('should render the AppHub container', () => {
    expect(wrapper.find(AppHub).length).toEqual(1);
    wrapper.setState({ isAuthenticated: true });
    expect(wrapper.find(AppHub).length).toEqual(1);
  });

  it('should render a loading message if not authenticated', () => {
    expect(wrapper.find(LoadingMessage).length).toEqual(1);
  });

  it('should render the Router (apps / content) if authenticated', () => {
    wrapper.setState({ isAuthenticated: true });
    expect(wrapper.find(Router).length).toEqual(1);
  });

  it('should pass the routes to the AppHub', () => {
    expect(wrapper.find(AppHub).prop('routes')).toEqual(routes);
    wrapper.setState({ isAuthenticated: true });
    expect(wrapper.find(AppHub).prop('routes')).toEqual(routes);
  });

  it('should pass the routes to the Router', () => {
    wrapper.setState({ isAuthenticated: true });
    expect(wrapper.find(Router).prop('routes')).toEqual(routes);
  });

  describe('handleDisplayRoutes', () => {
    let instance;
    beforeEach(() => {
      jest.resetAllMocks();
      localStorage.getItem = jest.fn(() => global.jwt.valid);
      wrapper = shallow(<AppHubRoot />);
      instance = wrapper.instance();
    });

    it('should clear the timeout when a valid token is found', () => {
      instance.handleDisplayRoutes();
      expect(global.clearTimeout).toHaveBeenCalled();
    });

    it('should set the user as being authenticated when a valid token is found', () => {
      instance.handleDisplayRoutes();
      expect(wrapper.state('isAuthenticated')).toEqual(true);
    });

    it('should set the timeout when there is no token or an invalid one', () => {
      localStorage.getItem = jest.fn(() => global.jwt.expired);
      global.setTimeout = jest.fn();
      wrapper = shallow(<AppHubRoot />);
      instance = wrapper.instance();
      instance.handleDisplayRoutes();
      expect(global.setTimeout).toHaveBeenCalled();
    });
  });
});
