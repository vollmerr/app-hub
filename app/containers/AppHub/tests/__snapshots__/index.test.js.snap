// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHub__Content /> should render correctly 1`] = `
.c0 {
  height: calc(100% - 50px);
}

<div
  className="c0"
/>
`;

exports[`<AppHub__Wrapper /> should render correctly 1`] = `
.c0 {
  height: 100%;
  overflow: hidden;
}

<CustomizedFabric
  className="c0"
/>
`;

exports[`<AppHubRoot /> should render correctly if authenticated 1`] = `
<AppHub__Wrapper>
  <withSaga(Connect(AppHub))
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "icon": "Apphub",
          "key": "home",
          "meta": Object {
            "desc": "Central location for CDT ODI applications",
            "keywords": "App, Hub",
            "title": "App Hub",
          },
          "name": "App Hub",
          "path": "/",
        },
        Object {
          "component": [Function],
          "exact": false,
          "icon": "Paas",
          "key": "paas",
          "meta": Object {
            "desc": "Personal Authorization Access System",
            "keywords": "tracking, access, authorization",
            "title": "PAAS",
          },
          "name": "PAAS",
          "path": "/paas",
        },
        Object {
          "component": [Function],
          "exact": false,
          "icon": "Spa",
          "key": "spa",
          "meta": Object {
            "desc": "Security Policy Acknowledgement",
            "keywords": "SPA",
            "title": "SPA",
          },
          "name": "SPA",
          "path": "/spa",
        },
        Object {
          "href": "https://cold.govops.ca.gov/",
          "icon": "COLD",
          "key": "cold",
          "meta": Object {
            "desc": "",
            "keywords": "",
            "title": "COLD",
          },
          "name": "COLD",
        },
        Object {
          "href": "http://bars.technology.ca.gov/",
          "icon": "BARS",
          "key": "bars",
          "meta": Object {
            "desc": "Badge Access Request System",
            "keywords": "badge, employee, new, student, request",
            "title": "BARS",
          },
          "name": "BARS",
        },
        Object {
          "href": "http://employeedirectory/",
          "icon": "ED",
          "key": "ed",
          "meta": Object {
            "desc": "Employee Directory",
            "keywords": "Employee, Directory, Lookup",
            "title": "ED",
          },
          "name": "ED",
        },
        Object {
          "component": [Function],
          "exact": false,
          "icon": "Demo",
          "key": "demo",
          "meta": Object {
            "desc": "Demo Description",
            "keywords": "Demo Keyword 1, Demo Keyword 2",
            "title": "Demo Title",
          },
          "name": "Demo",
          "path": "/demo",
        },
        Object {
          "component": [Function],
          "icon": "Loading",
          "key": "loading",
          "meta": Object {
            "desc": "test...",
            "keywords": "test, loading",
            "title": "TEST LOADING",
          },
          "name": "Test App Loading",
          "path": "/loading",
        },
      ]
    }
  />
  <AppHub__Content>
    <Router
      routes={
        Array [
          Object {
            "component": [Function],
            "exact": true,
            "icon": "Apphub",
            "key": "home",
            "meta": Object {
              "desc": "Central location for CDT ODI applications",
              "keywords": "App, Hub",
              "title": "App Hub",
            },
            "name": "App Hub",
            "path": "/",
          },
          Object {
            "component": [Function],
            "exact": false,
            "icon": "Paas",
            "key": "paas",
            "meta": Object {
              "desc": "Personal Authorization Access System",
              "keywords": "tracking, access, authorization",
              "title": "PAAS",
            },
            "name": "PAAS",
            "path": "/paas",
          },
          Object {
            "component": [Function],
            "exact": false,
            "icon": "Spa",
            "key": "spa",
            "meta": Object {
              "desc": "Security Policy Acknowledgement",
              "keywords": "SPA",
              "title": "SPA",
            },
            "name": "SPA",
            "path": "/spa",
          },
          Object {
            "href": "https://cold.govops.ca.gov/",
            "icon": "COLD",
            "key": "cold",
            "meta": Object {
              "desc": "",
              "keywords": "",
              "title": "COLD",
            },
            "name": "COLD",
          },
          Object {
            "href": "http://bars.technology.ca.gov/",
            "icon": "BARS",
            "key": "bars",
            "meta": Object {
              "desc": "Badge Access Request System",
              "keywords": "badge, employee, new, student, request",
              "title": "BARS",
            },
            "name": "BARS",
          },
          Object {
            "href": "http://employeedirectory/",
            "icon": "ED",
            "key": "ed",
            "meta": Object {
              "desc": "Employee Directory",
              "keywords": "Employee, Directory, Lookup",
              "title": "ED",
            },
            "name": "ED",
          },
          Object {
            "component": [Function],
            "exact": false,
            "icon": "Demo",
            "key": "demo",
            "meta": Object {
              "desc": "Demo Description",
              "keywords": "Demo Keyword 1, Demo Keyword 2",
              "title": "Demo Title",
            },
            "name": "Demo",
            "path": "/demo",
          },
          Object {
            "component": [Function],
            "icon": "Loading",
            "key": "loading",
            "meta": Object {
              "desc": "test...",
              "keywords": "test, loading",
              "title": "TEST LOADING",
            },
            "name": "Test App Loading",
            "path": "/loading",
          },
        ]
      }
    />
  </AppHub__Content>
</AppHub__Wrapper>
`;

exports[`<AppHubRoot /> should render correctly if not authenticated 1`] = `
<AppHub__Wrapper>
  <withSaga(Connect(AppHub))
    routes={
      Array [
        Object {
          "component": [Function],
          "exact": true,
          "icon": "Apphub",
          "key": "home",
          "meta": Object {
            "desc": "Central location for CDT ODI applications",
            "keywords": "App, Hub",
            "title": "App Hub",
          },
          "name": "App Hub",
          "path": "/",
        },
        Object {
          "component": [Function],
          "exact": false,
          "icon": "Paas",
          "key": "paas",
          "meta": Object {
            "desc": "Personal Authorization Access System",
            "keywords": "tracking, access, authorization",
            "title": "PAAS",
          },
          "name": "PAAS",
          "path": "/paas",
        },
        Object {
          "component": [Function],
          "exact": false,
          "icon": "Spa",
          "key": "spa",
          "meta": Object {
            "desc": "Security Policy Acknowledgement",
            "keywords": "SPA",
            "title": "SPA",
          },
          "name": "SPA",
          "path": "/spa",
        },
        Object {
          "href": "https://cold.govops.ca.gov/",
          "icon": "COLD",
          "key": "cold",
          "meta": Object {
            "desc": "",
            "keywords": "",
            "title": "COLD",
          },
          "name": "COLD",
        },
        Object {
          "href": "http://bars.technology.ca.gov/",
          "icon": "BARS",
          "key": "bars",
          "meta": Object {
            "desc": "Badge Access Request System",
            "keywords": "badge, employee, new, student, request",
            "title": "BARS",
          },
          "name": "BARS",
        },
        Object {
          "href": "http://employeedirectory/",
          "icon": "ED",
          "key": "ed",
          "meta": Object {
            "desc": "Employee Directory",
            "keywords": "Employee, Directory, Lookup",
            "title": "ED",
          },
          "name": "ED",
        },
        Object {
          "component": [Function],
          "exact": false,
          "icon": "Demo",
          "key": "demo",
          "meta": Object {
            "desc": "Demo Description",
            "keywords": "Demo Keyword 1, Demo Keyword 2",
            "title": "Demo Title",
          },
          "name": "Demo",
          "path": "/demo",
        },
        Object {
          "component": [Function],
          "icon": "Loading",
          "key": "loading",
          "meta": Object {
            "desc": "test...",
            "keywords": "test, loading",
            "title": "TEST LOADING",
          },
          "name": "Test App Loading",
          "path": "/loading",
        },
      ]
    }
  />
  <LoadingMessage />
</AppHub__Wrapper>
`;
